lib/core/src/server/config.js

```diff
@@ -27,30 +18,42 @@ function informAboutCustomConfig(defaultConfigName) {
   logger.info(`=> Using default webpack setup based on "${defaultConfigName}".`);
 }
 
-// `baseConfig` is a webpack configuration bundled with storybook.
-// Storybook will look in the `configDir` directory
-// (inside working directory) if a config path is not provided.
-export default options => {
+function getBabelConfig(options, presets) {
   const {
-    configType,
-    getBaseConfig,
     configDir,
-    defaultConfigName,
-    wrapInitialConfig = noopWrapper,
-    wrapBasicConfig = noopWrapper,
-    wrapDefaultConfig = noopWrapper,
+    defaultBabelConfig = devBabelConfig,
+    wrapDefaultBabelConfig = noopWrapper,
+    wrapBabelConfig = noopWrapper,
   } = options;
 
-  const babelOptions = {
+  const defaultConfig = wrapDefaultBabelConfig(defaultBabelConfig);
+  const customBabelConfig = loadCustomBabelConfig(configDir, defaultConfig);
+  const frameworkBabelConfig = wrapBabelConfig(customBabelConfig);
+  const presetBabelConfig = presets.extendBabel(frameworkBabelConfig);
+
+  return {
     // This is a feature of `babel-loader` for webpack (not Babel itself).
     // It enables a cache directory for faster-rebuilds
     // `find-cache-dir` will create the cache directory under the node_modules directory.
     cacheDirectory: findCacheDir({ name: 'react-storybook' }),
-    ...getBabelConfig(options),
+    ...presetBabelConfig,
   };
-  const baseConfig = getBaseConfig({ ...options, babelOptions });
-  const config = wrapInitialConfig(baseConfig, configDir);
+}
 
+function getWebpackConfig(options, babelOptions, presets) {
+  const {
+    configType,
+    getBaseConfig,
+    configDir,
+    defaultConfigName,
+    wrapInitialConfig = noopWrapper,
+    wrapBasicConfig = noopWrapper,
+    wrapDefaultConfig = noopWrapper,
+  } = options;
+
+  const baseConfig = getBaseConfig({ ...options, babelOptions, presets });
+  const initialConfig = wrapInitialConfig(baseConfig, configDir);
+  const config = presets.extendWebpack(initialConfig);
   const defaultConfig = wrapDefaultConfig(createDefaultWebpackConfig(config));
```

Comment:
We should remove these wrappers and rely on presets.

Reply:
In the long run - yes (Also wrote about it in Misc). But again, I would like to start using presets, without a huge refactoring in one PR.

Reply:
What I think we should do is instead of refactoring, we can support the new method of config without changing anything from the first. And once we are happy with the presets, we can remove the old one.